runtime/runtime.go

```diff
@@ -49,6 +50,21 @@ type Configuration struct {
 	// annonation on the running bundle is changed.
 	WatchBundle WatchRunningBundleFunc
 	ExtractedCredential
+	// StateMountLocation this is where on disk the state will be stored for a bundle
+	StateMountLocation string
+	// StateMasterNamespace the namespace where state created by bundles will be copied to between actions
+	StateMasterNamespace string
+}
+
+// StateManager defines an interface for managing state created by service bundles
+type StateManager interface {
```

Comment:
I would move this interface into the state file. this would follow the pattern that we have used for ExtractedCredential